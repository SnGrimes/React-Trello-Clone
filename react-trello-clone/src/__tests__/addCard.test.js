import React from 'react';
import { shallow} from 'enzyme'
import AddCard from '../components/addCard';

describe('addCard', () => {
  const mockAddCard = jest.fn();
  const mockCreateCard = jest.fn();
  const mockCancelCard = jest.fn();
  const props = {addCard: mockAddCard, cancelCard: mockCancelCard};
  const addCard = shallow(<AddCard {...props} />);
  jest.mock('../components/addCard', () => {
    return jest.fn().mockImplementation(() => {
      return {createCard: mockCreateCard };
    });
  });


  it('renders correctly', () => {
    expect(addCard).toMatchSnapshot();
  });
  it('initializes `state` with empty value', () => {
    expect(addCard.state().value).toEqual('');
  });
  describe('when typing into the card description input', () => {
    const description = 'This is a new card';
    beforeEach(() => {
      addCard.find('.new-card__desc-input').simulate('change', { target: { value: description}});
    });
    it('updates the description in `state`', () => {
      expect(addCard.state().value).toEqual(description);
    });
  }); //typing into card description input describe block
  describe('when clicking the `.new-card__button` button', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault')};
    beforeEach(() => {
      addCard.find('.new-card__form').simulate('submit',fakeEvent);
    });
    it('the createCard function is called', () => { 
      expect(mockAddCard).toHaveBeenCalled();
    });
  }); // click new card button describe block
  describe('when clicking the `x` button', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault')};
    beforeEach(() => {
      addCard.find('.new-card__x').simulate('click',fakeEvent);
    });
    it('cancels the creation of a new card', () => {
      expect(mockCancelCard).toHaveBeenCalledWith(false);
    });
  });
}); // addCard describe block
