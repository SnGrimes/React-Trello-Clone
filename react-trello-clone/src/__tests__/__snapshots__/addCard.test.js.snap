// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addCard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddCard
    addCard={[Function]}
    cancelCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="new-card__form"
        onSubmit={[Function]}
>
        <textarea
                className="new-card__desc-input"
                name="cardName"
                onChange={[Function]}
                size="30"
                value=""
        />
        <button
                className="new-card__button button success small"
                type="submit"
        >
                Save
        </button>
        <button
                className="new-card__x"
                onClick={[Function]}
        >
                <svg
                        className="feather feather-x"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                >
                        <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                        />
                        <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                        />
                </svg>
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            className="new-card__desc-input"
            name="cardName"
            onChange={[Function]}
            size="30"
            value=""
/>,
          <button
            className="new-card__button button success small"
            type="submit"
>
            Save
</button>,
          <button
            className="new-card__x"
            onClick={[Function]}
>
            <svg
                        className="feather feather-x"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
            >
                        <line
                                    x1="18"
                                    x2="6"
                                    y1="6"
                                    y2="18"
                        />
                        <line
                                    x1="6"
                                    x2="18"
                                    y1="6"
                                    y2="18"
                        />
            </svg>
</button>,
        ],
        "className": "new-card__form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "new-card__desc-input",
            "name": "cardName",
            "onChange": [Function],
            "size": "30",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "className": "new-card__button button success small",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="feather feather-x"
              fill="none"
              height="24"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
>
              <line
                            x1="18"
                            x2="6"
                            y1="6"
                            y2="18"
              />
              <line
                            x1="6"
                            x2="18"
                            y1="6"
                            y2="18"
              />
</svg>,
            "className": "new-card__x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
/>,
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
/>,
              ],
              "className": "feather feather-x",
              "fill": "none",
              "height": "24",
              "stroke": "currentColor",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": "2",
              "viewBox": "0 0 24 24",
              "width": "24",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "x1": "18",
                  "x2": "6",
                  "y1": "6",
                  "y2": "18",
                },
                "ref": null,
                "rendered": null,
                "type": "line",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "x1": "6",
                  "x2": "18",
                  "y1": "6",
                  "y2": "18",
                },
                "ref": null,
                "rendered": null,
                "type": "line",
              },
            ],
            "type": "svg",
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="new-card__form"
          onSubmit={[Function]}
>
          <textarea
                    className="new-card__desc-input"
                    name="cardName"
                    onChange={[Function]}
                    size="30"
                    value=""
          />
          <button
                    className="new-card__button button success small"
                    type="submit"
          >
                    Save
          </button>
          <button
                    className="new-card__x"
                    onClick={[Function]}
          >
                    <svg
                              className="feather feather-x"
                              fill="none"
                              height="24"
                              stroke="currentColor"
                              strokeLinecap="round"
                              strokeLinejoin="round"
                              strokeWidth="2"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                    >
                              <line
                                        x1="18"
                                        x2="6"
                                        y1="6"
                                        y2="18"
                              />
                              <line
                                        x1="6"
                                        x2="18"
                                        y1="6"
                                        y2="18"
                              />
                    </svg>
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <textarea
              className="new-card__desc-input"
              name="cardName"
              onChange={[Function]}
              size="30"
              value=""
/>,
            <button
              className="new-card__button button success small"
              type="submit"
>
              Save
</button>,
            <button
              className="new-card__x"
              onClick={[Function]}
>
              <svg
                            className="feather feather-x"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
              >
                            <line
                                          x1="18"
                                          x2="6"
                                          y1="6"
                                          y2="18"
                            />
                            <line
                                          x1="6"
                                          x2="18"
                                          y1="6"
                                          y2="18"
                            />
              </svg>
</button>,
          ],
          "className": "new-card__form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "new-card__desc-input",
              "name": "cardName",
              "onChange": [Function],
              "size": "30",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save",
              "className": "new-card__button button success small",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Save",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="feather feather-x"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
>
                <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                />
                <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                />
</svg>,
              "className": "new-card__x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <line
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
/>,
                  <line
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
/>,
                ],
                "className": "feather feather-x",
                "fill": "none",
                "height": "24",
                "stroke": "currentColor",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": "2",
                "viewBox": "0 0 24 24",
                "width": "24",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "x1": "18",
                    "x2": "6",
                    "y1": "6",
                    "y2": "18",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "line",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "x1": "6",
                    "x2": "18",
                    "y1": "6",
                    "y2": "18",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "line",
                },
              ],
              "type": "svg",
            },
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
